pkg/gds/models/v1/contacts.go

```diff
@@ -0,0 +1,202 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	pb "github.com/trisacrypto/trisa/pkg/trisa/gds/models/v1beta1"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	TechnicalContact      = "technical"
+	AdministrativeContact = "administrative"
+	LegalContact          = "legal"
+	BillingContact        = "billing"
+)
+
+type contactType struct {
+	contact *pb.Contact
+	kind    string
+}
+
+// Returns True if a Contact is not nil and has an email address.
+func ContactHasEmail(contact *pb.Contact) bool {
+	return contact != nil && contact.Email != ""
+}
+
+// Returns True if a Contact is verified.
+func ContactVerified(contact *pb.Contact) (verified bool, err error) {
+	if _, verified, err = GetContactVerification(contact); err != nil {
+		return false, err
+	}
+	return verified, nil
+}
+
+// Returns a standardized order of iterating through contacts.
+func contactOrder(contacts *pb.Contacts) []*contactType {
+	return []*contactType{
+		{contact: contacts.Technical, kind: TechnicalContact},
+		{contact: contacts.Administrative, kind: AdministrativeContact},
+		{contact: contacts.Legal, kind: LegalContact},
+		{contact: contacts.Billing, kind: BillingContact},
+	}
+}
+
+// Returns a function which iterates over the verified contacts in a Contacts object.
+func IterContacts(contacts *pb.Contacts, requireEmail bool) func() (*pb.Contact, string) {
```

Comment:
Could we please type the return function? E.g. 

```go
type ContactIterator func() (*pb.Contact, string)

func IterContacts(contacts *pb.Contacts, requireEmail bool) ContactIterator
```

This will make it a bit easier to read, but also to put a doc string above `ContactIterator` so that we can describe how to use it in a `for` loop.